### PR TITLE
chore: [FX-4186] update Modal to align with Design

### PR DESCRIPTION
**Pull request by dmaklygin**
[FX-4186]

### Description

* full-screen size property has to take 100% of width and height of the screen (not 100% - Nrem as it is now)

### How to test

- use [temploy](https://picasso.toptal.net/fx-4186-fix-modal/?path=/story/overlays-modal--modal) to check all states in all screen sizes

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Ensure that deployed demo has expected results and good examples
- [x] Self reviewed

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4186]: https://toptal-core.atlassian.net/browse/FX-4186?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dmaklygin** [comment]
@toptal-anvil ping reviewers

File: packages/picasso/src/Modal/styles.ts [hunk 1]
```diff
@@ -40,16 +50,19 @@ export default ({ screens, sizes }: Theme) =>
       width: '50rem',
     },
     'full-screen': {
-      height: maxHeight,
-      width: maxWidth,
+      height: '100%',
+      width: '100%',
+      maxWidth: '100%',
+      maxHeight: '100%',
```
Comment (sashuk) [inline packages/picasso/src/Modal/styles.ts:56]: NIT: Maybe to keep order of `height` and `width` properties the same (before it was `maxHeight` and then `maxWidth`)?

File: packages/picasso/src/Modal/styles.ts [hunk 2]
```diff
@@ -19,15 +17,27 @@ export default ({ screens, sizes }: Theme) =>
...
-      [screens('xs')]: {
-        maxHeight: maxHeightForExtraSmall,
-        maxWidth: maxWidthForExtraSmall,
-        margin: '1rem',
+      [screens('sm', 'md', 'lg', 'xl')]: {
+        margin: '2rem',
+        maxHeight,
+      },
+      [screens('sm')]: {
+        maxWidth: '26rem',
```
Comment (sashuk) [inline packages/picasso/src/Modal/styles.ts:31]: Is it possible to postpone these changes (as https://toptal-core.atlassian.net/wiki/spaces/Base/pages/3353608586/Modal is not final)? It is still unclear what to do in the following case:

```
user has screen width of 1450px (xl breakpoint)
the modal is 1200px wide
user slightly changes screen width to 1390px (lg breakpoint)
the modal shrinks to 872px right away
```

Right now the transition described above happens gradually, so there is a chance that there will be no need in `max-width` for modals as the gradual shrink (as it works now) will be preferred

**Comment by sashuk** [comment]
`full-screen` changes work as expected!

https://github.com/toptal/picasso/assets/1390758/bc4d0806-1388-4c9e-be4c-c11376dd1875

File: packages/picasso/src/Modal/styles.ts (hunk 2 above)
Comment (sashuk) [reply packages/picasso/src/Modal/styles.ts:31]: Actually, these changes should be avoided for now, as later there will be five different modal sizes and the `maxWidth` will not depend on the screen size, but on `Modal.size` property (the way it works now). The change will be addressed in https://toptal-core.atlassian.net/browse/FX-4211

Comment (dmaklygin) [reply packages/picasso/src/Modal/styles.ts:31]: Yeah, I'll remove those changes then

**Review by sashuk — APPROVED** [review]
Code looks good, previously tested on temploy!

**Review by mkrl — APPROVED** [review]
Great job!

**Pull request merged**